client/customloaders/customfileloader/index.js

```diff
@@ -0,0 +1,59 @@
+var Promise = require('bluebird');
+var fse = Promise.promisifyAll(require('fs-extra'));
+var loaderUtils = require('loader-utils');
+var mime = require('mime');
+var path = require('path');
+
+module.exports = function (content) {
+  this.cacheable && this.cacheable();
+  if (!this.emitFile) {
+    throw new Error('emitFile is required from module system');
+  }
+  var query = loaderUtils.parseQuery(this.query);
+  var url = loaderUtils.interpolateName(this, query.name || '[hash].[ext]', {
```

Comment:
Those are webpack's default handles for renaming files.
So the default '[hash].[ext]' will change the filename 'someImage.jpg' to '345938yr8f378645387453857.jpg'.